src/trans.c

```diff
@@ -1500,16 +1500,24 @@ Obj FuncON_KERNEL_ANTI_ACTION (Obj self, Obj ker, Obj f, Obj n) {
   UInt    deg, i, j, rank, len;
   Obj     out;
 
-  if (INT_INTOBJ(ELM_LIST(ker, LEN_LIST(ker))) == 0) {
+  assert(IS_LIST(ker));
+  assert(IS_INTOBJ(n));
+
+  len = LEN_LIST(ker);
+  if (len == 1 && INT_INTOBJ(ELM_LIST(ker, 1)) == 0) {
```

Comment:
`ker` is a "flat" kernel of a transformation or `[0]`, the flat kernel of a transformation consists of strictly positive integers, so it should never be the case that when `ker` has length greater than 1, that it contains `0`. 

To reiterate, the previous version of the function was wrong, so I think we want the behaviour to be different now.